Paracha_Junaid/Assignments/Flask_SQL/friends/server.py

```diff
@@ -0,0 +1,96 @@
+
+from flask import Flask, render_template, session, request, redirect, flash
+import re
+from mysqlconnection import MySQLConnector
+
+EMAIL_REGEX = re.compile(r'^[a-zA-Z0-9.+_-]+@[a-zA-Z0-9._-]+\.[a-zA-Z]+$')
+PW_REGEX = re.compile(r'^(?=.*[0-9])(?=.*[A-Z])\w{8,}$')
+app = Flask(__name__)
+app.secret_key = 'my_secret_key'
+mysql = MySQLConnector(app, 'friends')
+
+@app.route('/')
+def index():
+    login_user = mysql.query_db("SELECT first_name, last_name, email, users.id, DATE_FORMAT(users.created_at,'%d/%m/%Y %h:%i %p') as created_at FROM users")
+    
+    return render_template('index.html', login_user = login_user)
+
+@app.route('/show', methods = ['POST'])
+def friends_index():
+    session['login_id'] = request.form['login_user']   
+    session['users_info'] = mysql.query_db("SELECT u2.id as id, u2.email as User_Name, u2.first_name as First_Name, u2.last_name as Last_Name, DATE_FORMAT(users.created_at,'%d/%m/%Y %h:%i %p') as created_at FROM friends.users LEFT JOIN friendship as f ON f.user_id=users.id LEFT JOIN users as u2 ON u2.id =f.Friend_id WHERE f.User_id='"+session['login_id'] +"'")
+
+    return redirect('/')  
+
+@app.route('/friend', methods = ['POST'])
+def add_friend():    
+    if len(request.form['email']) < 1 or len(request.form['first_name']) < 1 or len(request.form['last_name']) < 1:
```

Comment:
separate out the validation fields even though they are checking for the same thing